google/resource_dataproc_job.go

```diff
@@ -0,0 +1,1068 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/dataproc/v1"
+)
+
+func resourceDataprocJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDataprocJobCreate,
+		Read:   resourceDataprocJobRead,
+		Delete: resourceDataprocJobDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			// Ref: https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.jobs#JobReference
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "global",
+				ForceNew: true,
+			},
+
+			// If a job is still running, trying to delete a job will fail. Setting
+			// this flag to true however will force the deletion by first cancelling
+			// the job and then deleting it
+			"force_delete": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+				ForceNew: true,
```

Comment:
I wonder whether it makes sense to make this updatable, in case the user doesn't realize the attribute exists (or forgets to set it) and then later realizes they want to force delete the job. What do you think?

Reply:
fair enough, added